dev/ProjectReunion_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "MddBootstrap.h"
+
+#include "IDynamicDependencyLifetimeManager.h"
+
+STDAPI AddFrameworkToPath(PCWSTR  path);
+
+IDynamicDependencyLifetimeManager* g_lifetimeManager = nullptr;
+DLL_DIRECTORY_COOKIE g_dllDirectoryCookie = 0;
+wil::unique_cotaskmem_string g_frameworkPath;
+
+STDAPI MddBootstrapInitialize(
+    const CLSID& appDynamicDependencyLifetimeManager)
+{
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager> lifetimeManager;
+    RETURN_IF_FAILED(CoCreateInstance(appDynamicDependencyLifetimeManager, nullptr, CLSCTX_LOCAL_SERVER, __uuidof(IDynamicDependencyLifetimeManager), reinterpret_cast<void**>(lifetimeManager.addressof())));
+
+    wil::unique_cotaskmem_string path;
+    RETURN_IF_FAILED(lifetimeManager->GetFrameworkPath(path.addressof()));
+
+    RETURN_IF_FAILED(AddFrameworkToPath(path.get()));
+
+    g_lifetimeManager = lifetimeManager.detach();
```

Comment:
Probably should add a `FAILFAST_IF(g_lifetimeManager)` here to shake out multiple initialization. Or make calling this method clearly idempotent.

Reply:
Oh also, check out wil's init_once support to make that even magicer.

Reply:
It's a 1-time thing for main() so I don't see multiple initialization. I've added FAIL_FAST traps at the start of init as a sanity check in case someone oddly enough calls this twice

init_once is a bit too magical here. Maybe refactor someday but I'd like to keep things a bit simpler for now as we try to get past the crawl stage